coffee/lib/abstract-chosen.coffee

```diff
@@ -178,13 +179,29 @@ class AbstractChosen
         option.search_text = if option.group then option.label else option.html
 
         unless option.group and not @group_search
-          option.search_match = this.search_string_match(option.search_text, regex)
+          option.search_match = this.search_string_match(this.normalize_search_text(option.search_text), regex)
           results += 1 if option.search_match and not option.group
 
           if option.search_match
             if searchText.length
-              startpos = option.search_text.search zregex
-              text = option.search_text.substr(0, startpos + searchText.length) + '</em>' + option.search_text.substr(startpos + searchText.length)
+              startpos = this.normalize_search_text(option.search_text).search zregex
+              poslength = this.normalize_search_text(searchText).length
+
+              # If any normalization changed the search_text, perform a better
+              # logic to highlight results
+              if this.normalize_search_text(option.search_text) != option.search_text
```

Comment:
you should normalize the search_text only once (instead of doing it many times for each option) and the option search_text only once per option, to make things faster in case the normalization is a bit complex. It will be faster this way.